### ISSUE TITLE
My config causes a crash, what am I doing wrong?

### ISSUE THREAD

**Issue by sanctas** [ESPresense/ESPresense-companion#338]
Config: 

```
# MQTT Connection, if empty will query and use hassio provided mqtt
mqtt:
  host: homeassistant.local
  port: 1883
  ssl: false
  username: my_username
  password: my_password

# This gets added to the x,y,z to derive a gps location
gps:
  latitude: 38.897957
  longitude: -77.036560
  elevation: 146

# How long before device considered stale
timeout: 30
# How long before device is considered away
away_timeout: 120

optimization:
  enabled: true
  interval_secs: 60
  limits:
    absorption_min: 2.5
    absorption_max: 3.5
    tx_ref_rssi_min: -70
    tx_ref_rssi_max: -50
    rx_adj_rssi_min: -15
    rx_adj_rssi_max: 20

weighting:
  algorithm: gaussian
  props:
    sigma: 0.10

# Floors w/ the points to draw it in meters
floors:
  - id: first
    name: First Floor
    # Bounds (x,y,z) of map in meters
    bounds: [[0, 0, 0], [18.5, 19.2, 1.5]]
    rooms:
      - name: Garage
        points:
          - [0, 0]
          - [5.5, 0]
          - [5.5, 6]
          - [0, 6]
      - name: Kitchen
        points:
          - [0, 6]
          - [0, 13.2]
          - [0.5, 13.2]
          - [1, 13.5]
          - [2.75, 13.5]
          - [3.25, 13.2]
          - [3.75, 13.2]
          - [3.75, 6]
      - name: Living Room
        points:
          - [3.75, 6]
          - [3.75, 11.5]
          - [9.5, 11.5]
          - [9.5, 5]
          - [5.5, 5]
          - [5.5, 6]
      - name: Suite Living Room
        points: 
          - [3.75, 11.5]
          - [6.75, 11.5]
          - [6.75, 17.25]
          - [3.75, 17.25]
      - name: Suite Bedroom
        points:
          - [6.75, 14.5]
          - [9.5, 14.5]
          - [9.5, 17.25]
          - [6.75, 17.25]
      - name: Suite Bathroom
        points:
          - [6.75, 14.5]
          - [9.5, 14.5]
          - [9.5, 11.5]
          - [6.75, 11.5]
      - name: Foyer
        points: 
          - [5.5, 5]
          - [9, 5]
          - [9, 2.1]
          - [5.5, 2.1]
      - name: Hallway
        points:
          - [9, 5]
          - [11.2, 5]
          - [11.2, 6.25]
          - [12.2, 6.25]
          - [12.2, 4]
          - [9, 4]
      - name: Guest Bathroom
        points:
          - [9.5, 5]
          - [9.5, 8.75]
          - [11.2, 8.75]
          - [11.2, 5]
      - name: Master Bathroom
        points:
          - [9.5, 8.75]
          - [9.5, 12.2]
          - [11.2, 12.2]
          - [11.2, 8.75]
      - name: Master Bedroom
        points:
          - [11.2, 12.2]
          - [15.7, 12.2]
          - [15.7, 5.5]
          - [12.2, 5.5]
          - [12.2, 6.25]
          - [11.2, 6.25]
      - name: Guest Bedroom
        points:
          - [12.2, 5.5]
          - [15.7, 5.5]
          - [15.7, 0]
          - [12.2, 0]
      - name: Office
        points:
          - [9, 4]
          - [12.2, 4]
          - [12.2, 0]
          - [9, 0]

# Locations of espresense nodes in meters
nodes:
  - name: Living Room
    point: [9.2, 9.5]
    floors: ["first"]
  - name: Office
    point: [10.2, 0.1]
    floors: ["first"]
  - name: Kitchen
    point: [0.1, 7.75]
    floors: ["first"]
  - name: Suite Living Room
    point: [3.85, 16]
    floors: ["first"]
  - name: Master Bedroom
    point: [13.1, 12]
    floors: ["first"]
# Devices to track
devices:
  - id: myname_s_iphone_15_pro_ble
    name: MyName's iPhone 15 Pro BLE
  - id: wifename_s_iphone
    name: WifeName's iPhone BLE
```

Error Log:

```
[17:37:38 INF] Loading /config/espresense/config.yaml
Unhandled exception. System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at ESPresense.Models.Node.Update(Config c, ConfigNode cn, IEnumerable`1 floors) in /App/Models/Node.cs:line 48
   at ESPresense.Models.State.<.ctor>g__LoadConfig|0_1(Config c) in /App/Models/State.cs:line 26
   at ESPresense.Models.State..ctor(ConfigLoader cl) in /App/Models/State.cs:line 68
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.ConstructorInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(ServiceCallSite callSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(ServiceCallSite callSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitIEnumerable(IEnumerableCallSite enumerableCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(ServiceCallSite callSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.Resolve(ServiceCallSite callSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.CreateServiceAccessor(Type serviceType)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.GetService(Type serviceType, ServiceProviderEngineScope serviceProviderEngineScope)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService(IServiceProvider provider, Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService[T](IServiceProvider provider)
   at Microsoft.Extensions.Hosting.Internal.Host.StartAsync(CancellationToken cancellationToken)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.RunAsync(IHost host, CancellationToken token)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.RunAsync(IHost host, CancellationToken token)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.Run(IHost host)
   at Program.<Main>$(String[] args) in /App/Program.cs:line 95
```

**Comment by CoMPaTech** [comment]
Seems you are missing the 'height' of your nodes (e.g. third locator) - you only reference two points instead of three.

i.e.
```
nodes:
  - name: Living Room
    point: [9.2, 9.5]
    floors: ["first"]
```

Should be like
```
nodes:
  - name: Living Room
    point: [9.2, 9.5, 0.5]
    floors: ["first"]
```

**Comment by DTTerastar** [comment]
This is fixed in 0.4.3!
